staging/src/k8s.io/csi-translation-lib/plugins/gce_pd.go

```diff
@@ -77,10 +77,10 @@ func translateAllowedTopologies(terms []v1.TopologySelectorTerm) ([]v1.TopologyS
 			var newExp v1.TopologySelectorLabelRequirement
 			if exp.Key == v1.LabelZoneFailureDomain {
 				newExp = v1.TopologySelectorLabelRequirement{
-					Key:    GCEPDTopologyKey,
+					Key:    key,
 					Values: exp.Values,
 				}
-			} else if exp.Key == GCEPDTopologyKey {
+			} else if exp.Key == key {
 				newExp = exp
 			} else {
 				return nil, fmt.Errorf("unknown topology key: %v", exp.Key)
```

Comment:
Question for @davidz627 what happens today if a user specifies a non-zone label in allowedtopologies?

Reply:
`return nil, fmt.Errorf("unknown topology key: %v", exp.Key)`, not sure if that answers your question? Fail fast :+1:

Reply:
I mean, when migration is off. Is this something that in-tree volumes allow users to do today?

Reply:
I tested with a storage class with unknown topology key and the pvc is stuck in pending